bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/DanfossAirUnit.java

```diff
@@ -0,0 +1,278 @@
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.danfossairunit.internal;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.net.InetAddress;
+import java.nio.charset.StandardCharsets;
+import java.time.DateTimeException;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DanfossAirUnit} class represents the air unit device and build the commands to be sent by
+ * {@link DanfossAirUnitCommunicationController}
+ *
+ * @author Ralf Duckstein - Initial contribution
+ * @author Robert Bach - heavy refactorings
+ */
+
+@SuppressWarnings("SameParameterValue")
+@NonNullByDefault
+public class DanfossAirUnit implements Commands {
+
+    private final Logger logger = LoggerFactory.getLogger(DanfossAirUnit.class);
+    private final DanfossAirUnitCommunicationController communicationController;
+
+    public DanfossAirUnit(InetAddress inetAddr, int port) {
+        this.communicationController  = new DanfossAirUnitCommunicationController(inetAddr, port);
+    }
+
+    public void cleanUp() {
+        this.communicationController.disconnect();
+    }
+
+    private boolean getBoolean(byte[] operation, byte[] register) throws IOException {
+        return communicationController.sendRobustRequest(operation, register)[0] != 0;
+    }
+
+    private void setSetting(byte[] register, boolean value) throws IOException {
+        setSetting(register, value ? (byte) 1 : (byte) 0);
+    }
+
+    private short getWord(byte[] operation, byte[] register) throws IOException {
+        byte[] resultBytes = communicationController.sendRobustRequest(operation, register);
+        return (short)((resultBytes[0] << 8)  | (resultBytes[1] & 0xFF));
+    }
+
+    private byte getByte(byte[] operation, byte[] register) throws IOException {
+        return communicationController.sendRobustRequest(operation, register)[0];
+    }
+
+    private String getString(byte[] operation, byte[] register) throws IOException {
+        // length of the string is stored in the first byte
+        byte[] result = communicationController.sendRobustRequest(operation, register);
+        return new String(result, 1, result[0], StandardCharsets.US_ASCII);
```

Comment:
Not UTF-8? Unusual.

Reply:
Same as above. Right now working as expected  (the devices unit name is reported correctly), but no real documentation.